www/main/share-sync.js

```diff
@@ -19,26 +24,44 @@ module.exports = {
   },
   async tryConnection() {
     const host = store.get('userId');
+    let syncCode = null;
 
-    if (window.namespaceString) {
-      return window.namespaceString;
-    }
-    try {
-      const result = await request(`${SYNC_API_URL}/sync/begin/${host}`);
-      const {
-        data: { namespaceString },
-      } = JSON.parse(result);
-
-      if (window.io !== undefined) {
-        window.namespaceString = namespaceString;
-        onConnect(namespaceString);
-      } else {
-        // TODO: Wait for io or something
+    const getNewCode = async () => {
```

Comment:
does this function need to be in tryConnection?

Reply:
No, it doesn't need to be.